snc.sh

```diff
@@ -137,7 +134,19 @@ OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="$(curl -l "${MIRROR}/${OPENSHIFT_RELEA
 export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
 echo "Setting OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
 
+# Extract openshift-install binary if not present in current direcory
+if ! which $OPENSHIFT_INSTALL; then
+    echo ${OPENSHIFT_PULL_SECRET} > pull-secret
+    baremetal_installer_image=$(oc adm release -a pull-secret info ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE} --image-for=baremetal-installer)
+    oc image -a pull-secret extract ${baremetal_installer_image} --confirm --path /usr/bin/openshift-install:.
+    chmod +x openshift-install
+    rm pull-secret
```

Comment:
Not sure it should go to current directory, or to some subdirectory. We need to set OPENSHIFT_INSTALL to the path to the installer after doing this though, as the `which` test could fail because `OPENSHIFT_INSTALL` was set to some wrong value.

Reply:
Seeing .gitignore, we also use the current directory for the generated ssh keys. Maybe it's a good opportunity to start using `$INSTALL_DIR/snc_data/` and put the pull secret and the installer there?

Reply:
Actually, one advantage of not putting it in ./openshift-install is that snc will fetch the image/the binary every time if ./openshift-install does not exist, which ensures we are not using an outdated binary. But that can be improved later if you prefer.

Reply:
@cfergeau even you put an `$INSTALL_DIR/snc_data` then a user need to delete this directory before caching another installer binary right or you mean to say that then irrespective to data in the directory we are always going to rewrite the installer binary? In the current form if I want to test a different installer binary with different payload I can just compile the installer binary and put it in snc directory so the extraction process ignore it (This would be good for debugging) but we might lose that if we always force to rewrite installer binary.

Reply:
```
OPENSHIFT_INSTALL=${OPENSHIFT_INSTALL:-./openshift-install} 
if ! which $OPENSHIFT_INSTALL; then 
# put installer in somewhere/openshift-install
# OPENSHIFT_INSTALL=somewhere/openshift-install
fi
```
if OPENSHIFT_INSTALL is set/./openshift_install exist, they are going to be used. If they don't exist, then the installer will always be extracted from the baremetal image.
After we've tested this PR for a while, we probably can remove the default value for OPENSHIFT_INSTALL